### DISCUSSION TITLE
Interactive ORBAT

### DISCUSSION THREAD

**Discussion by ThomasHalwax** [orbat-mapper/orbat-mapper#50] [Ideas]
Hi Kjell,

this is @ThomasHalwax, member of the [ODIN](https://github.com/syncpoint/ODIN) development team. Your ORBAT chart looks great and would be a fantastic addition to ODIN (and other non-public research projects we are working on). Do you have plans to make the chart interactive, i.e. users may rearrange elements?

My vision has always been to have an ORBAT-as-a-service where different apps can have different views on the ORBAT. The entities should be addressable via a http REST interface and users might address different versions of the ORBAT (to reflect the changes over time). Ahh, so many ideas and so litte time :-)

Thank you for your work & best regards
Thomas

**Comment by kjellmf** [comment]
Hi Thomas,

Thanks for your interest in my hobby project. Making the ORBAT chart editable is on my todo-list. Until now I have primarily experimented with rendering and layout. Progress has been slow and creating a smart layout algorithm is not trivial. I'm currently leaning towards a more pragmatic approach where the user easily can modify the layout by rearranging and adjusting the elements instead of doing everything automatically. 

The idea behind ORBAT mapper is to create a tool for recreating historic battles and military operations (with simple playback functionality).  
As you mention, ORBATs typically changes over time. I have not figured out a good way to model this yet. An ORBAT-as-a-service sounds like a good idea.  If you need some inspiration, take a look at [OE Data Integration Network](https://odin.tradoc.army.mil/) (also called ODIN :-) If you poke around you can find some nice examples of ORBATs (for instance https://odin.tradoc.army.mil/FS/Caucasus/Ariana/SUPREME%20HIGH%20COMMAND)

Regards

**Reply by ThomasHalwax** [reply]
Hi Kjell,

thank you for the links. Funny to see that others chose ODIN for their project names, too.

Regarding the changes over time: I've been experimenting with [isomorphic git](https://isomorphic-git.org) and was surprised how well it worked. Maybe you can model the aspect of time as a branch or tag of your root objects, where every tag is assigned to a point-in-time.

Best regards, Thomas

**Reply by kjellmf** [reply]
Interesting idea to use the git model for tracking history. I'll keep that in mind. Thanks!
